### DISCUSSION TITLE
Nested Navigation. Add a Redirect?

### DISCUSSION THREAD

**Discussion by wcrowther** [posva/unplugin-vue-router#304] [Questions and Help] [answered]
I am working on a sample Vue app that I am switching from regular **vue-router** to  **unplugin-vue-router** that looks like this:
 
![image](https://github.com/posva/unplugin-vue-router/assets/6372711/fe1c952d-eff3-4c51-a17a-a89e421d84e7)

Here is the new page structure that I trying to make work. 

![image](https://github.com/posva/unplugin-vue-router/assets/6372711/b43af911-91f3-484c-b076-3a4cf59102e1)

The navigation works ok but the problem is that if I make the nested **how.vue** file **index.vue** it always shows as selected because the link to it is **/About**. If I leave it as **how.vue** then the top-level **About** button just shows the page but does not pre-select a child component (ie: how, why, where). My existing **vue-router** implementation has a redirect at **/About** to **/About/How**  that solves the problem.

My best guess is to add the following code to router.js to make a redirect:

```
const router = createRouter(
{
	history: createWebHistory(),
	extendRoutes(routes) {
		routes.push( { path: '/About', name: 'AboutRedirect', redirect: { path: '/About/How' }} )
	} 
})
```
but that does not work and throws the error below. I saw a discussion that suggestions this may be a bug or is there some syntax that I am missing. Any ideas how to make this work?

![image](https://github.com/posva/unplugin-vue-router/assets/6372711/45794bbf-a93e-4425-a81e-5f42a7ee227d)

**Comment by wcrowther** [comment]
I am still curious if it is possible to add a route programmatically with extendRoutes() as above without getting an error but I figured out a solution to my useCase. Instead of above, I added a redirect in the /About page that redirects to /About/How and solves the problem with the selected tab not applying active class correctly. 

```
import { definePage } from 'vue-router/auto'
	
definePage({
	redirect: '/About/How'
})
```

**Comment by posva** [comment, marked as answer]
> The navigation works ok but the problem is that if I make the nested how.vue file index.vue it always shows as selected because the link to it is /About.

Use the exact active class instead of the active one to set the style

**Reply by whitersun** [reply]
Hello posva, I still dont understand how to code with exact active class instead of the active. Does have any example code or something to show?

**Reply by posva** [reply]
https://router.vuejs.org/guide/essentials/active-links.html#Exact-active-links
